tests/end_to_end/test_vulnerability_detector/conftest.py

```diff
@@ -161,33 +279,53 @@ def setup(preconditions, teardown, host_manager) -> Generator[Dict, None, None]:
 
     logging.info("Running setup")
     result = {}
+    failed_agents = []
 
     if preconditions:
         logging.info("Running preconditions")
-        result = launch_parallel_operations(preconditions['tasks'], host_manager)
+        target_to_ignore = []
+        agents_to_check = host_manager.get_group_hosts("agent")
+
+        if 'target_os' in preconditions:
+            agents_to_check = filter_hosts_by_os(host_manager, preconditions['target_os'])
+            target_to_ignore = list(set(host_manager.get_group_hosts('agent')) - set(agents_to_check))
+
+        result = launch_parallel_operations(preconditions, host_manager, target_to_ignore)
         logging.info(f"Preconditions finished. Results: {result}")
 
-        for host in result.keys():
-            if result[host]['checks']['all_successfull'] is False:
-                logging.critical(f"Test failed for host {host}. Check logs for more information")
-                logging.critical(f"Evidences: {result[host]['evidences']}")
+        logging.info(f"Result of preconditions: {result}")
+
+        test_timestamp = datetime.datetime.utcnow()
```

Comment:
This function is deprecated and the work has already been done in a previous PR where it was updated to the expected one.

Reply:
Done dfb9045be1dff6f4fbd201c602ff25447f957f3d